userspace/libscap/linux/scap_procs.c

```diff
@@ -1377,3 +1377,15 @@ int32_t scap_linux_get_threadlist(struct scap_platform* platform, struct ppm_pro
 	}
 	return SCAP_SUCCESS;
 }
+
+int32_t scap_linux_get_fdlist(struct scap_platform* platform, struct scap_threadinfo *tinfo, char *lasterr)
+{
+	uint64_t num_fds_ret = 0;
+	char proc_dir[SCAP_MAX_PATH_SIZE];
+	struct scap_ns_socket_list* sockets_by_ns = NULL;
+	struct scap_linux_platform* linux_platform = (struct scap_linux_platform*)platform;
+
+	snprintf(proc_dir, sizeof(proc_dir), "%s/proc/%d/", scap_get_host_root(), tinfo->pid);
```

Comment:
i would add a comment here, to explain why we use the `pid` and not the `tid`, so something like "we collect the fds only for the main thread"